lib/cog/command/pipeline/executor.ex

```diff
@@ -215,7 +232,38 @@ defmodule Cog.Command.Pipeline.Executor do
         fail_pipeline(state, :binding_error, "Error preparing to execute command pipeline '#{state.request["text"]}': #{msg}")
       bound ->
         {current_bound, bound_scope} = collect_bound(bound)
-        {:next_state, :get_options, %{state | current_bound: current_bound, scope: bound_scope}, 0}
+        {:next_state, :maybe_expand, %{state | current_bound: current_bound, scope: bound_scope}, 0}
+    end
+  end
+
+  @doc """
+  `expand` -> {:stop, :shutdown} | {:next_state, :get_options}
+
+  If a command begins with 'user' or 'site' it is treated as an alias and expanded.
+  """
+  def maybe_expand(:timeout, %__MODULE__{current: %{command: "user:" <> _user_alias}=current_bound}=state),
+    do: expand_alias(current_bound, state)
+  def maybe_expand(:timeout, %__MODULE__{current: %{command: "site:" <> _site_alias}=current_bound}=state),
+    do: expand_alias(current_bound, state)
+  def maybe_expand(:timeout, state),
+    do: {:next_state, :get_options, state, 0}
+
+  defp expand_alias(current_bound, state) do
+    case AliasExpander.expand(current_bound.command) do
+      {:ok, %Ast.Pipeline{invocations: invocations}} ->
+        cond do
+          state.current_depth + 1 > state.max_depth ->
+            Helpers.send_reply("Error expanding alias. I'm trying to can... but I just can't.", state.request, state.mq_conn)
+            fail_pipeline(state, :alias_expansion_error, "Error expanding alias. Too much recursion.")
+          true ->
+            [new_current|rest] = invocations
+            # We insert a marker here to determine how deep we have gone with expansion
+            rest = rest ++ [%Marker{alias_name: current_bound.command}]
```

Comment:
I add a marker to determine how deep we have gone with alias expansion. When we expand an alias we add the marker to the stack before the new invocation list and increase the current_depth by 1. When we come across the marker later during execution, we drop it and decrease the current_depth by 1. If our current_depth ever exceeds our max_depth we shutdown and return an error to the user.